### PR TITLE
Save formatted text for faster cipher generation

### LINKED ISSUE #18: Cache formatted books to speed up cipher generation
- [ ] Save formatted texts (like the plain books - perhaps replace?) 
- [ ] Check for cached formatted texts (again perhaps instead of plain books)
- [ ] Simply take slice from formatted text

### PR DESCRIPTION
**Pull request by Jatewo**
This pull request refactors and improves the cipher generation pipeline, focusing on text formatting, error handling, and code maintainability. The main changes include moving and enhancing the text formatting logic, updating the cipher generation API, and improving tests for edge cases and correctness.

**Text formatting improvements:**
* Moved and refactored the `format_text` function from `Fetcher` to `utils/formatting.py`, ensuring all book text is consistently formatted before use. The new implementation converts numbers to words, removes non-alphabetic characters, and normalizes accents. [[1]](diffhunk://#diff-5a05d3bbb81cf5e7980a07bf7461da70813f41e76847caf6e66975496f0010f8L3-R11) [[2]](diffhunk://#diff-d659cb7430704b17fc3f11ef0fc89ca4747027b7e94617463a7ffd03c3795ea4R1) [[3]](diffhunk://#diff-d659cb7430704b17fc3f11ef0fc89ca4747027b7e94617463a7ffd03c3795ea4L9-R35)
* Updated `Fetcher` to format book text immediately after fetching and before caching, ensuring cached books are always in the correct format. [[1]](diffhunk://#diff-5a05d3bbb81cf5e7980a07bf7461da70813f41e76847caf6e66975496f0010f8L112-R120) [[2]](diffhunk://#diff-5a05d3bbb81cf5e7980a07bf7461da70813f41e76847caf6e66975496f0010f8L134-R156)

**API and logic changes:**
* Changed the `generate_cipher` function signature to explicitly require `min_len` and `max_len` parameters instead of a single length, making the API clearer and more flexible. [[1]](diffhunk://#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312L2-R4) [[2]](diffhunk://#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312L14-R28)
* Improved error handling during cipher generation, catching and reporting errors related to invalid plaintext and book slice length. [[1]](diffhunk://#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312L14-R28) [[2]](diffhunk://#diff-5a05d3bbb81cf5e7980a07bf7461da70813f41e76847caf6e66975496f0010f8L175-L198)

**Testing improvements:**
* Updated and expanded tests to cover new formatting logic, error cases (such as non-string input and invalid plaintext), and edge cases in slicing and fetching book text. Added tests for the new `format_text` function and removed outdated tests from `Fetcher`. [[1]](diffhunk://#diff-c0469c5748e9772e6e098a1b279aceb9b03b066fb8f42cf9a818416fdc1bb607L40-L64) [[2]](diffhunk://#diff-c0469c5748e9772e6e098a1b279aceb9b03b066fb8f42cf9a818416fdc1bb607L77-R53) [[3]](diffhunk://#diff-c0469c5748e9772e6e098a1b279aceb9b03b066fb8f42cf9a818416fdc1bb607L86-L91) [[4]](diffhunk://#diff-c0469c5748e9772e6e098a1b279aceb9b03b066fb8f42cf9a818416fdc1bb607R80) [[5]](diffhunk://#diff-c0469c5748e9772e6e098a1b279aceb9b03b066fb8f42cf9a818416fdc1bb607L113-R93) [[6]](diffhunk://#diff-c0469c5748e9772e6e098a1b279aceb9b03b066fb8f42cf9a818416fdc1bb607L192-R178) [[7]](diffhunk://#diff-aa6ac68474cbc30f59e1c17b9d89198d26e4bf31504e443045cc032b598c6cbdL1-R60) [[8]](diffhunk://#diff-959a1effd1cb0e25016584deef9be1994af65814f318346bf3a63801b5db1636R4-R60)

**Code organization and cleanup:**
* Removed duplicated and obsolete formatting logic from `Fetcher`, centralizing it in `utils/formatting.py`. Updated imports accordingly. [[1]](diffhunk://#diff-5a05d3bbb81cf5e7980a07bf7461da70813f41e76847caf6e66975496f0010f8L3-R11) [[2]](diffhunk://#diff-d659cb7430704b17fc3f11ef0fc89ca4747027b7e94617463a7ffd03c3795ea4R1) [[3]](diffhunk://#diff-c0469c5748e9772e6e098a1b279aceb9b03b066fb8f42cf9a818416fdc1bb607R5)
* Improved test fixtures and cleanup logic to ensure test isolation and reliability.

These changes make the codebase more robust, maintainable, and easier to extend for future features.

### CODE REVIEW

**Review by TheMOHD3886 — APPROVED** [review]
nice bruva 👍

File: src/utils/formatting.py [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
-	def replace_number(match):
-		"""Replace a number in the text with its word representation.
-
-		Args:
-			match (re.Match): The regex match object containing the number.
-
-		Returns:
-			str: The word representation of the number.
-		"""
+	def replace_number(match: re.Match) -> str:
```
Comment (Mafusn) [inline src/utils/formatting.py]: No definition 😢

File: test/text-fetching/test_fetcher.py [hunk 2]
```diff
@@ -83,12 +59,15 @@ def test_slicing_short_text(short_text):
     min_len = 100
     assert len(short_text) < min_len
     with pytest.raises(ValueError) as excinfo:
-        fetcher.get_random_book_slice(book_text=short_text, min_len=min_len)
-    assert "book_text is shorter than the specified max_len" in str(excinfo.value)
+        fetcher.get_random_book_slice(
+            book_text=short_text, min_len=min_len, max_len=200
+        )
+    assert "book_text is shorter than the specified max_le" in str(
```
Comment (Mafusn) [inline test/text-fetching/test_fetcher.py]: Sorry, but it's missing "n" 🐥

**Review by Mafusn — APPROVED** [review]
LGTM, only silly changes 🦖

**Review by Mafusn — APPROVED** [review]
Very nice 🥇

**Review by TheMOHD3886 — APPROVED** [review]
lgtm again boss 👍
